Dualstack configuration procedure (#66646)
openshift/openshift-docs: modules/nw-multus-configure-dualstack-ip-address.adoc

// Module included in the following assemblies:
//
// * networking/multiple_networks/configuring-additional-network.adoc

:_content-type: PROCEDURE 

[id="nw-multus-configure-dualstack-ip-address_{context}"]
= Creating a configuration for assignment of dual-stack IP addresses dynamically

Dual-stack IP address assignment can be configured with the `ipRanges` parameter for:

> * IPv4 addresses

Reviewer: You need an empty line here - this is not showing up in the preview as a bulleted list.

Author: ty

* IPv6 addresses
* multiple IP address assignment 

.Procedure

> . Set type to whereabouts.

Reviewer: ```suggestion
. Set `type` to `whereabouts`.
```

> . Use `ipRanges` to allocate IP addresses.

Reviewer: ```suggestion
. Use `ipRanges` to allocate IP addresses as shown in the following example:
```

Reviewer: ^normally we need introductory text for examples

> 

Reviewer: You need a + sign here to indent the example.

Author: Thank you

> 

Reviewer 2: ```suggestion
+
```

> + [source,yaml]

Reviewer 2: ```suggestion
[source,yaml]
```

cniVersion: operator.openshift.io/v1
kind: Network
metadata:
  name: cluster
spec:
  additionalNetworks:
  - name: whereabouts-shim
    namespace: default
    type: Raw
    rawCNIConfig: |-
      {
       "name": "whereabouts-dual-stack",
       "cniVersion": "0.3.1,
       "type": "bridge",
       "ipam": {
         "type": "whereabouts",
         "ipRanges": [
                  {"range": "192.168.10.0/24"},
                  {"range": "2001:db8::/64"}
              ]
       }
      }

----

> . Attach network to a pod. For more information, see "Adding a pod to an additional network".

Reviewer: Did you intend to restart numbering here because it's a new section? If so, I think there needs to be an introductory sentence like, "After allocating the IP addresses, perform the following steps:"

Author: no I did not...

. Verify that all IP addresses are assigned.

> Run the following command to ensure the IP adresses are assigned as meta-data.

Reviewer: ```suggestion
. Run the following command to ensure the IP addresses are assigned as metadata.
```

Reviewer: I think this last one should be a step? And then indent the commands below using `+`? I'm kind of confused because the preview shows "run the following two commands" - maybe it just hasn't updated yet. But there is a typo in `addresses` and `metadata`.

Author: Should only say run the following command

> Run the following two commands to ensure the IP adresses are assigned as meta-data.

Reviewer 3: there's a single command below.

furthermore, it depends on the workload image - i.e. it will fail is the customer the image is using does not have the `ip` command baked in.

Finally, I don't understand the "assigned as meta-data" part of the sentence.

Author: @dougbtv can you answer the "assigned as meta-data" part?

Author: Fixed the two command bit- does the meta-data part need to change?

> 

Reviewer 2: ```suggestion
+
```

[source,yaml]
----

> oc exec -it mypod -- ip a

Reviewer 2: ```suggestion
$ oc exec -it mypod -- ip a
```

Author: yep

// Module included in the following assemblies:
//
// * networking/multiple_networks/configuring-additional-network.adoc

:_content-type: PROCEDURE 

[id="nw-multus-configure-dualstack-ip-address_{context}"]
= Creating a configuration for assignment of dual-stack IP addresses dynamically

Dual-stack IP address assignment can be configured with the `ipRanges` parameter for:
* IPv4 addresses
* IPv6 addresses
* multiple IP address assignment 

.Procedure

. Set type to whereabouts.
. Use `ipRanges` to allocate IP addresses.

[source,yaml]
----
cniVersion: operator.openshift.io/v1
kind: Network
metadata:
  name: cluster
spec:
  additionalNetworks:
  - name: whereabouts-shim

>     namespace: default

Reviewer 4: Need add `type: Raw`

Author: done

Reviewer 5: @weliang1 can you confirm this change is correctly implemented?

Reviewer 4: @nmuller66 One more thing.
Could you update `{"range": "192.168.10.1/24"},` to `{"range": "192.168.10.0/24"},`?

Author: @weliang1 absolutely yes

Author: @weliang1 with this update, is that a LGTM from a QE perspective
